### ISSUE TITLE
Verify keys properly

### ISSUE THREAD

**Issue by david415** [coyim/coyim#494]
tl;dr - "CoyIM has completely failed thus far to provide cryptographic assurances to it's users because keys aren't verified.". If you are pro-human rights violations then continue not fixing this problem as hard as you can.

Reproduce this bad behavior:

step 1. Alice and Bob add each other as contacts in their XMPP+OTR clients. Alice uses CoyIM. They exchange fingerprints and their comm channel is now verified in both directions. Alice's CoyIM knows what fingerprint Bob's public key is supposed to have.

step 2. Bob and Alice can chat and everything appears to work and is encrypted.

step 3. Bob changes his key and doesn't tell Alice.

step 4. Bob and Alice chat again. Alice's CoyIM client doesn't show any kind of alert or warning to indicate that her contact's key has changed.

Is there anything ambiguous about this?

Thanks and cheers!

**Comment by david415** [comment]
oops this is a duplicate of https://github.com/coyim/coyim/issues/480

**Comment by olabini** [comment]
Hi. Thank for your report - this is obviously not good behavior. In order to triage and fix, it would be very helpful if you could provide some more information. Specifically:
- What client is Bob using?
- In what way does Bob change his key?
- How are the fingerprints marked/verified in Coy on Alice's side?
- Which released version of Coy are you using, when seeing this behavior?

**Comment by olabini** [comment]
Will work on this in conjunction with #480 - same updates apply.
